### ISSUE TITLE
Support model defaults on creatish methods

### ISSUE THREAD

**Issue by bobobo1618** [SectorLabs/django-postgres-extra#20]
Sorry about the title.

The problem is if I have a model like this:

```python
class Thing(psqlextra.models.PostgresModel):
    field = models.TextField(blank=True)
```

And I do this:

```python
Thing.objects.upsert(
    conflict_target=['pk'],
    fields={'pk': 0, 'field': None})
```

I get:

`django.db.utils.IntegrityError: null value in column "field" violates not-null constraint`

While if I simply do `Thing.objects.create(pk=0, field=None)`, everything works fine, as Django internally turns the `None` into an empty string.

**Comment by Photonios** [comment]
Thanks for the report! Seems you found an edge case. An interesting one :-)

It seems as though the code forgets to call some method on the text field that turns None into an empty string.

I'll have a look at the code later today. This should be relatively easy to fix.

**Comment by elnygren** [comment]
Sort of related: upsert doesn't support `DEFAULT VALUES` at all. This is a problem in a case where:
* a field may not be NULL during creation
* the field must not be updated if the upsert becomes an UPDATE

Leaving the field out from `fields={}` means that UPDATEs work but INSERTs fail. Having the field in `fields={'thefield': 'default'}` makes INSERTs work but UPDATE will overwrite, which was not wanted.

**Comment by Photonios** [comment]
@bobobo1618, I've tried the following test case:

```
def test_on_conflict_text_field():

    model = get_fake_model(
        {
            'name': models.TextField(blank=True)
        },
        model_base=models.Model
    )

    obj2 = (
        model.objects
        .create(pk=0, name=None)
    )
```

And it doesn't result in Django turning `None` into an empty string:

```
django.db.utils.IntegrityError: null value in column "name" violates not-null constraint
DETAIL:  Failing row contains (0, null).
```

In an effort to come to the same results you're getting, I've been reading through the Django source code: https://github.com/django/django/blob/dc63ad7ac09ef694f19ae1d7496367614889e6fd/django/db/models/fields/__init__.py#L2093

I don't see any evidence that Django would turn `None` into `''` and I can't confirm it. I've even opened a completely fresh and new Django project and declared a model exactly as you described. It still throws the same exception.

If this behaviour is really expected and I am doing something wrong, please let me know. However, your example did point out two interesting bugs related to using `pk` to refer to the primary key. See #23 and #24.

@elnygren, I am looking into the issue with the defaults now :)

**Comment by Photonios** [comment]
@elnygren The issue has been fixed, see release v1.13:

https://github.com/SectorLabs/django-postgres-extra/releases/tag/v1.13
